src/test/scala/is/hail/utils/IntervalSuite.scala

```diff
@@ -0,0 +1,211 @@
+package is.hail.utils
+
+import is.hail.annotations.ExtendedOrdering
+import is.hail.expr.types.TInt32
+import org.testng.annotations.Test
+
+class IntervalSuite {
+
+  val pord: ExtendedOrdering = TInt32().ordering
+
+  // set of intervals chosen from 5 endpoints spans the space of relations
+  // that two non-empty intervals can have with each other.
+  val points: IndexedSeq[Int] = 1 to 5
+
+  val test_intervals: IndexedSeq[SetInterval] =
+    for {
+      s <- points
+      e <- points
+      is <- Array(true, false)
+      ie <- Array(true, false)
+    } yield SetInterval(s, e, is, ie)
+
+  val test_itrees: IndexedSeq[SetIntervalTree] =
+    for {
+      i1 <- test_intervals
+      i2 <- test_intervals
+    } yield SetIntervalTree(Array(i1, i2).zipWithIndex)
```

Comment:
Are we sure two is enough to test the code?  This would be a good case for code coverage, basically I want to check all the code paths in IntervalTreeNode.

At the least we should test 0, and 1, too.